bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/binding/jrubyscripting/internal/JRubyScriptEngineFactory.java

```diff
@@ -0,0 +1,119 @@
+    private final static Set<String> INSTANCE_PRESETS = Set.of();
+    private final static Set<String> GLOBAL_PRESETS = Set.of("scriptExtension", "automationManager", "ruleRegistry",
+            "items", "voice", "rules", "things", "events", "itemRegistry", "ir", "actions", "se", "audio",
+            "lifecycleTracker");
+
+    private final javax.script.ScriptEngineFactory factory = new org.jruby.embed.jsr223.JRubyEngineFactory();
+
+    // formatter turned off because it does not appropriately format chained streams
+    //@formatter:off
+    private final List<String> scriptTypes = Stream.concat(factory.getExtensions().stream(), 
+                                                           factory.getMimeTypes().stream())
+                                                   .collect(Collectors.toUnmodifiableList());
+    //@formatter:on
+
+    // Adds @ in front of a set of variables so that Ruby recogonizes them as instance variables
+    private static Map.Entry<String, Object> mapInstancePresets(Map.Entry<String, Object> entry) {
+        if (INSTANCE_PRESETS.contains(entry.getKey())) {
+            return Map.entry("@" + entry.getKey(), entry.getValue());
+        } else {
+            return entry;
+        }
+    }
+
+    // Adds $ in front of a set of variables so that Ruby recogonizes them as global variables
+    private static Map.Entry<String, Object> mapGlobalPresets(Map.Entry<String, Object> entry) {
+        if (GLOBAL_PRESETS.contains(entry.getKey())) {
+            return Map.entry("$" + entry.getKey(), entry.getValue());
+        } else {
+            return entry;
+        }
+    }
+
+    // The activate call is activate binding and set the bindings configuration
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> config) {
+        configuration.update(config, factory);
+    }
+
+    // The modified call updates configuration for binding
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        configuration.update(config, factory);
+    }
+
+    @Override
+    public List<String> getScriptTypes() {
+        return scriptTypes;
+    }
+
+    @Override
+    public void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValues) {
+
+        // formatter turned off because it does not appropriately format chained streams
+        //@formatter:off
+        Map<String, Object> filteredScopeValues =
+                              scopeValues
+                             .entrySet()
+                             .stream()
+                             .filter(map -> !FILTERED_PRESETS.contains(map.getKey()))
+                             .map(JRubyScriptEngineFactory::mapInstancePresets)
+                             .map(JRubyScriptEngineFactory::mapGlobalPresets)
+                             .collect(Collectors.toMap(map -> map.getKey(), map -> map.getValue()));
+        //@formatter:on
```

Comment:
I would prefer to keep the formatter always on and help it by adding `//` at the end of line instead. That way you get the same readable code but with the added benefit that it will still follow the global indentation rules (# of tabs/spaces). It also prevents issues whenever you forget to turn the formatter back on.

Reply:
Done